### PR TITLE
Set an issue to Unassigned

### PR DESCRIPTION
**Pull request by benjidea**
Allow 'null' as assignee to set the issue to 'unassigned'.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description.
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests.
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description.
- [x] Reviewed the code.
- [x] Verified that the appropriate tests have been written or valid explanation given.
- [x] If applicable, tested by installing this plugin on the Jenkins instance.

### CODE REVIEW

File: src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java [hunk 1]
```diff
@@ -87,14 +87,13 @@ protected Execution(final AssignIssueStep step, final StepContext context)
```
Comment (nrayapati) [inline src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java]: Change to this class alone is sufficient to un-assign issue. I don't think we need to introduce the custom class - AssigneeParameter. Please remove and try to test it.

File: hugo/content/changelog.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ lastmodifierdisplayname = "Naresh Rayapati"
 +++
```
Comment (nrayapati) [inline hugo/content/changelog.md]: can you also update docs on AddComment step at  https://github.com/jenkinsci/jira-steps-plugin/tree/master/hugo/content/steps/comment please

File: src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java (hunk 1 above)
Comment (benjidea) [reply src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java]: Unfortunately the ImmutableMap can't handle null values. Do you have another idea?

File: hugo/content/changelog.md (hunk 2 above)
Comment (benjidea) [reply hugo/content/changelog.md]: Of course! I will add an example.
You mean 'Assign Issue' for this pull request?

**Comment by benjidea** [comment]
Thanks for your review!

File: src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java (hunk 1 above)
Comment (nrayapati) [reply src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java]: Good catch!, I forgot about that part. lets try the normal map then. 

```java
  public ResponseData<Void> assignIssue(final String issueIdorKey, final String userName) {
    try {
      Map input = Maps.newHashMap();
      input.put("name", userName);
      return parseResponse(
          jiraEndPoints
              .assignIssue(issueIdorKey, input)
              .execute());
    } catch (Exception e) {
      return buildErrorResponse(e);
    }
  }
```

Comment (benjidea) [reply src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AssignIssueStep.java]: Good idea! It works the way it should.

**Comment by nrayapati** [comment]
🎉 Thanks a lot for this change @benjidea, I will release it soon after further local testing.

**Pull request merged**